modules/installation-cloudformation-control-plane.adoc

```diff
@@ -0,0 +1,386 @@
+// Module included in the following assemblies:
+//
+// * installing/installing_aws_upi/installing-aws-upi.adoc
+
+[id="installation-cloudformation-control-plane-{context}"]
+= CloudFormation template for control plane machines
+
+You can use the following CloudFormation template to deploy the control plane
+machines that you need for your {product-title} cluster.
+
+[source,yaml]
+----
+AWSTemplateFormatVersion: 2010-09-09
+Description: Template for Openshift Cluster UPI Node Launch (EC2 master instances)
+
+Parameters:
+  InfrastructureName:
+    AllowedPattern: ^([a-zA-Z][a-zA-Z0-9\-]{0,31})$
+    MaxLength: 32
+    MinLength: 1
+    ConstraintDescription: Infrastructure name must be alphanumeric, start with a letter and a maximum of 32 characters
+    Description: A short, unique cluster ID used to tag cloud resources and identify items owned/used by the cluster.
+    Type: String
+  RhcosAmi:
+    Description: Current RHEL CoreOS AMI to use for boostrap
+    Type: AWS::EC2::Image::Id
+  AutoRegisterDNS:
+    Default: "yes"
+    AllowedValues:
+    - "yes"
+    - "no"
+    Description: Do you want to invoke DNS etcd registration (requires Hosted Zone info provided)?
+    Type: String
+  PrivateHostedZoneId:
+    Description: The Route53 private zone ID to register the etcd targets with (e.g Z21IXYZABCZ2A4)
+    Type: String
+  PrivateHostedZoneName:
+    Description: The Route53 zone to register the targets with (No trailing dot - e.g. cluster.mycorp.com)
+    Type: String
+  Master0Subnet:
+    Description: The subnets (recommend private) to launch the master nodes into
+    Type: AWS::EC2::Subnet::Id
+  Master1Subnet:
+    Description: The subnets (recommend private) to launch the master nodes into
+    Type: AWS::EC2::Subnet::Id
+  Master2Subnet:
+    Description: The subnets (recommend private) to launch the master nodes into
+    Type: AWS::EC2::Subnet::Id
+  MasterSecurityGroupId:
+    Description: The master security group ID to associate with master nodes.
+    Type: AWS::EC2::SecurityGroup::Id
+  IgnitionLocation:
+    Default: https://api.$CLUSTER_NAME.$DOMAIN:22623/config/master
+    Description: Location to fetch bootstrap ignition from. (Recommend to use the autocreated ignition config location.)
+    Type: String
+  CertificateAuthorities:
```

Comment:
Certificate Authority x.509 certificate is stored unencrypted and available to all users including AWS support.  This absolutely cannot happen.

We have to think of another way.

Reply:
> Certificate Authority x.509 certificate is stored unencrypted and available to all IAM users in the account including AWS support. This absolutely cannot happen.

this is the trust certificate to trust the server that will serve the ignition. what's the problem with this being available to other users...?

Reply:
Just to underline @abhinavdahiya's point, this is the certificate, not the key.  The keys get pushed up to S3, but aren't exposed in the CloudFormation template.  I don't see a problem with leaking certificates.

Reply:
Keep in mind the cert is in the wild in at least two places.  1) S3 ... since the bucket and object both have no encryption for data at rest the data is replicated throughout the AWS S3 ecosystem. It is unknown who has access to this ecosystem.  2) CloudFormation ... The data is exposed as a parameter value.  The data is replicated throughout the AWS CloudFormation backend.  It is unknown who has access to this ecosystem.

I'm less concerned about the IAM users of the account having access than AWS world wide support possibly having access.

Reply:
> I'm less concerned about the IAM users of the account having access than AWS world wide support possibly having access.

But again, what is your concern with a public certificate at all?  Say a malicious Mallory acquires the cert.  What are you concerned about him doing with it.  He can't impersonate the Ignition-config server unless he *also* has the key, so leaking certs doesn't seem like a problem to me.

Reply:
Since #1203 is changing the s3 block_public_* things to true I'm less unhappy about this.